### DISCUSSION TITLE
Provisioning notes

### DISCUSSION THREAD

**Discussion by emilekberg** [grafana/grafana#44968] [Dashboards, Panels & Visualizations]
It would be nice if i could track changes to the provisioned dashboard using notes and potentially updated by. 

the use case is that we store our jsons as code and deploy along side of our application. In test environment and such, it would be nice to verify the correct version has been deployed by appending the branch name or whatever as a note. 

To my testing, i cannot see how i could do this, or any documentation in doing so.
If dashboards was updated manually it would be fine, but i guess no version for the provisioned dashboard exists?

**Comment by Mario-Hofstaetter** [comment]
I tried exporting a dashboard and just now noticed that the version change history is omitted from the JSON.

It could be easily implemented to **optionally** include the desired version history + notes in the JSON, so it gets imported / included when provisioning?

Would that be sufficient for your case @emilekberg  ?

Maybe this is already possible if the correct JSON keys and schema are known?
<https://grafana.com/docs/grafana/latest/administration/provisioning/#dashboards> does not document the supported schema for dashboard JSON during provisioning.

The http api docs <https://grafana.com/docs/grafana/latest/http_api/dashboard_versions/> show the schema for versions, but it is not clear if a provisioned JSON may also included versions.

Example with pseudo json below. The version info could easily created by code (git commits, whatever)

![grafik](https://user-images.githubusercontent.com/33002073/155079224-2f505579-0f7c-4783-b44f-e2ab800b4fa3.png)

`Dashboard-Versions-Mockup.json`:
```json
{
  "title": "My New Dashboard",
  "uid": "a6NoXbf7z",
  "version": 4,
  "versions": [
    {
      "version": 1,
      "date": "2022-02-22 07:52:00",
      "updatedBy": "admin",
      "notes": "Initial save"
    },
    {
      "version": 4,
      "date": "2022-02-22 07:56:49",
      "updatedBy": "admin",
      "notes": "Deployed from branch origin/dev git hash 8331bcc57246c2c0a294dcbf92a9..."
    }
  ]
}
```

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
